rfcs/56-better-reflect.md

```diff
@@ -0,0 +1,318 @@
+# bevy_reflect: `Reflect` as a unique `Reflect`
+
+## Summary
+
+- create a new trait: `PartialReflect`
+- `Reflect: PartialReflect`
+- All methods on `Reflect` that **does not** depend on the uniqueness of the
+  underlying type are moved to `PartialReflect`
+- Most things should now accept a `dyn PartialReflect` rather than
+  `dyn Reflect`.
+- It is possible to convert a `PartialReflect` into a `Reflect` using a
+  `into_full` method.
+- `FromReflect` becomes `FromPartialReflect`.
+
+## Jargon
```

Comment:
Could "pass_for" be re-added to the jargon? It was useful terminology, imo, for referencing the represented type.